src/transaction_builder.js

```diff
@@ -183,6 +219,209 @@ export class TransactionBuilder {
     return this;
   }
 
+  /**
+   * If you want to prepare a transaction which will become valid at some point
+   * in the future, or be invalid after some time, you can set a timebounds
+   * precondition. Internally this will set the `minTime`, and `maxTime`
+   * preconditions. Conflicts with `setTimeout`, so use one or the other.
+   *
+   * @param {Date|number} minEpochOrDate  Either a JS Date object, or a number
+   *     of UNIX epoch seconds. The transaction is valid after this timestamp.
+   *     Can't be negative. If the value is `0`, the transaction is valid
+   *     immediately.
+   * @param {Date|number} maxEpochOrDate  Either a JS Date object, or a number
+   *     of UNIX epoch seconds. The transaction is valid until this timestamp.
+   *     Can't be negative. If the value is `0`, the transaction is valid
+   *     indefinitely.
+   *
+   * @returns {TransactionBuilder}
+   */
+  setTimebounds(minEpochOrDate, maxEpochOrDate) {
+    // Force it to a date type
+    if (typeof minEpochOrDate === 'number') {
+      minEpochOrDate = new Date(minEpochOrDate * 1000);
+    }
+    if (typeof maxEpochOrDate === 'number') {
+      maxEpochOrDate = new Date(maxEpochOrDate * 1000);
+    }
+
+    if (this.timebounds !== null) {
+      throw new Error(
+        'TimeBounds has been already set - setting timebounds would overwrite it.'
+      );
+    }
+
+    // Convert that date to the epoch seconds
+    const minTime = Math.floor(minEpochOrDate.valueOf() / 1000);
+    const maxTime = Math.floor(maxEpochOrDate.valueOf() / 1000);
+    if (minTime < 0) {
+      throw new Error('min_time cannot be negative');
+    }
+    if (maxTime < 0) {
+      throw new Error('max_time cannot be negative');
+    }
+
+    this.timebounds = { minTime, maxTime };
+
+    return this;
+  }
+
+  /**
+   * If you want to prepare a transaction which will only be valid within some
+   * range of ledgers, you can set a ledgerbounds precondition.
+   * Internally this will set the `minLedger` and `maxLedger` preconditions.
+   *
+   * @param {number} minLedger  The minimum ledger this transaction is valid at
+   *     or after. Cannot be negative. If the value is `0` (the default), the
+   *     transaction is valid immediately.
+   *
+   * @param {number} maxLedger  The maximum ledger this transaction is valid
+   *     before. Cannot be negative. If the value is `0`, the transaction is
+   *     valid indefinitely.
+   *
+   * @returns {TransactionBuilder}
+   */
+  setLedgerbounds(minLedger, maxLedger) {
+    if (this.ledgerbounds !== null) {
+      throw new Error(
+        'LedgerBounds has been already set - setting ledgerbounds would overwrite it.'
+      );
+    }
+
+    if (minLedger < 0) {
+      throw new Error('min_ledger cannot be negative');
+    }
+    if (maxLedger < 0) {
+      throw new Error('max_ledger cannot be negative');
+    }
+
+    if (minLedger > maxLedger) {
```

Comment:
If minLedger != 0 and maxLedger == 0, an exception will be thrown here.

Reply:
Nice catch, thanks.